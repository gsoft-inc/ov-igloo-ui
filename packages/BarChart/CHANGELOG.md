# @igloo-ui/bar-chart

## 0.3.1

### Patch Changes

- 9739155: Gave ColorPicker the new Wokleap look. Color now uses css variables to update its color. Fixed spacing issues in tooltip. Updated hopper token version.

## 0.3.0

### Minor Changes

- 4096378: Added the rebrand. Added disabled prop to Popover to work with StackedBar.

## 0.2.3

### Patch Changes

- a41e1d4: Fix: moved destructuring of props to function parameters so that storybook will automatically show the default value in the props table. It was done in Modal because it needed to send props to useDialog. In this case, @default annotation was used in comments.

## 0.2.2

### Patch Changes

- e1fd1ac: Fix from BarChart can have some improvements #296 issue
  Fix space between the label and the bar
  Fix glitch when the barGraph is reloading
  Fix loading animation speed

## 0.2.1

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console

## 0.2.0

### Minor Changes

- 0b51874: add animation

## 0.1.0

### Minor Changes

- defc58d: initial release of BarChart
