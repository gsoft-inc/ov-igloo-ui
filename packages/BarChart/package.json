{
  "name": "@igloo-ui/bar-chart",
  "version": "0.4.3",
  "main": "dist/BarChart.js",
  "module": "dist/BarChart.js",
  "types": "dist/BarChart.d.ts",
  "styles": "dist/bar-chart.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Franck franck.gaudin@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/tokens": "^2.1.0",
    "@hopper-ui/tokens": "^4.2.0",
    "classnames": "^2.3.2",
    "framer-motion": "^6.5.1"
  },
  "devDependencies": {
    "@types/lodash.camelcase": "^4.3.7"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
