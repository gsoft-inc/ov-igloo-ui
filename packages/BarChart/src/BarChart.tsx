import * as React from 'react';
import cx from 'classnames';

import { useTransition, animated, easings } from 'react-spring';

import './bar-chart.scss';

export interface DataSet {
  /** Add label text above the bar chart. */
  label: string;
  /** The value displayed beside the bar chart. */
  value: number;
  /** Add the color to the bar chart. */
  color?: string;
}

export interface BarChartProps extends React.ComponentProps<'li'> {
  /** The data object that will be used for the bar chart. */
  dataSet: DataSet[];
  /** Add a data-test tag for automated tests. */
  dataTest?: string;
  /** Add a specific class to the component. */
  className?: string;
}

const BarChart: React.FunctionComponent<BarChartProps> = (
  props: BarChartProps
) => {
  const { dataSet, className, dataTest, ...rest } = props;

  if (!dataSet) {
    return null;
  }

  const setWidth = (value: number): undefined | string => {
    if (!value || value === 0) {
      return undefined;
    }

    const values = dataSet.map((data) => data.value);
    const maxValue = Math.max(...values);

    const size = (value * 100) / maxValue;

    return `${size}%`;
  };

  const transitions = useTransition(dataSet, {
    from: { width: '0' },
    enter: (item: DataSet) => [
      {
        width: setWidth(item.value),
      },
    ],
    config: { duration: 500, easing: easings.easeInCubic },
  });

  const barChart = transitions(({ width }, item: DataSet) => {
    const { label, value, color } = item;

    const animateWidth = value > 0 ? width : undefined;

    return (
      <li key={`ids-bar-chart-${value}`} className="ids-bar-chart" {...rest}>
        <span className="ids-bar-chart__label">{label}</span>
        <div className="ids-bar-chart__content">
          <animated.div
            className="ids-bar-chart__graph"
            data-value={value}
            style={{
              width: animateWidth,
              backgroundColor: color,
            }}
          />
          <span className="ids-bar-chart__value">{value}</span>
        </div>
      </li>
    );
  });

  // const barChart = dataSet.map((data: DataSet) => {
  //   const { label, value, color } = data;
  //
  //   return (
  //     <li key={`ids-bar-chart-${value}`} className="ids-bar-chart" {...rest}>
  //       <span className="ids-bar-chart__label">{label}</span>
  //       <div className="ids-bar-chart__content">
  //         <div
  //           className="ids-bar-chart__graph"
  //           data-value={value}
  //           style={{
  //             width: setWidth(value),
  //             background: color,
  //           }}
  //         />
  //         <span className="ids-bar-chart__value">{value}</span>
  //       </div>
  //     </li>
  //   );
  // });

  return (
    <ul
      className={cx('ids-bar-chart__container', className)}
      data-test={dataTest}
    >
      {barChart}
    </ul>
  );
};

export default BarChart;
