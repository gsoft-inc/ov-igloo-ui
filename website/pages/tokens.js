import { useEffect, useState } from 'react';
import { useRouter } from 'next/router';
import cx from 'classnames';

import Link from 'next/link';
import pkg from '@igloo-ui/tokens/package.json';
import Display from '../components/Display';
import Title from '../components/Title';
import Code from '../components/Code';
import Tag from '../components/Tag';
import TokenList from '../components/TokenList';

import iconToken from '../svg/token.svg';

export default function Tokens(props) {
  const { data, categories } = props;
  const { asPath } = useRouter();

  const [activeLink, setActiveLink] = useState();
  const [baseToken, setBaseToken] = useState('10');
  const [styleToken, setStyleToken] = useState('scss');

  const handleTokenChange = (event) => {
    setBaseToken(event.currentTarget.value);
  };

  const handleStyleChange = (event) => {
    setStyleToken(event.currentTarget.value);
  };

  useEffect(() => {
    setActiveLink(window.location.hash);
  }, [asPath]);

  const tokensList = categories.map((key) => {
    const dataType = baseToken === '10' ? 'base10' : 'default';

    const token = data[dataType].filter(
      (obj) => Object.keys(obj).toString() === key
    );
    return (
      <TokenList
        key={key}
        data={token[0][key]}
        title={key}
        options={{ base: baseToken, style: styleToken }}
      />
    );
  });

  const tokensNav = categories.map((category, index) => {
    return (
      <li key={index.toString()}>
        <Link href={`/tokens#${category}`}>
          <a className={cx(activeLink === `#${category}` && 'is-active')}>
            {category}
          </a>
        </Link>
      </li>
    );
  });

  return (
    <>
      <div className="io-section io-section--grid io-section--hasAside">
        <div className="io-aside">
          <ul className="io-subnav">{tokensNav}</ul>
        </div>
        <div className="io-tokens">
          <Display icon={iconToken}>
            <div className="io-display__content">
              Tokens{' '}
              <Tag appearance="ghost" className="io-display__tag">
                v{pkg.version}
              </Tag>
            </div>
          </Display>
          <Title level={2} as="h2">
            Getting started
          </Title>
          <p>
            To install <Code inline>@igloo-ui/tokens</Code> in your project, you
            will need to run the following command using{' '}
            <a href="npmjs.org" target="_blank">
              npm:
            </a>
          </p>
          <Code language="bash">{`npm install @igloo-ui/icons`}</Code>
          <p>If you prefer Yarn, </p>
          <Code language="bash">{`yarn add @igloo-ui/icons`}</Code>
          <Title level={3} as="h3">
            Usage
          </Title>
          <p>In Sass, you can import the files by doing:</p>
          <Code language="jsx">
            {`// Fonts${'\n'}@import '@igloo-ui/tokens/dist/_fonts.scss';${'\n'}// SCSS${'\n'}@import '@igloo-ui/tokens/dist/_variables.scss';${'\n'}// CSS${'\n'}@import '@igloo-ui/tokens/dist/_colors.scss';${'\n'}`}
          </Code>
          <Title level={3} as="h3">
            Base 10 support
          </Title>
          <p>
            If your project uses the mathematical trick of basing the value of{' '}
            <Code inline>1rem</Code> equals <Code inline>10px</Code>, there is
            also an <strong>base10</strong> output within the dist folder of
            this package.
          </p>
          <p>In CSS, you can import the files by doing:</p>
          <Code>@import '~@igloo-ui/tokens/dist/base10/variables.css';</Code>

          <div className="io-tokens__list">
            <div className="io-options io-options--large">
              <Title level={3} as="h3" className="io-options__title">
                Options
              </Title>
              <select
                className="io-select"
                value={styleToken}
                onChange={handleStyleChange}
              >
                <option selected value="scss">
                  scss
                </option>
                <option value="css">css</option>
              </select>
              <select
                className="io-select"
                value={baseToken}
                onChange={handleTokenChange}
              >
                <option selected value="10">
                  base 10
                </option>
                <option value="16">base 16</option>
              </select>
            </div>
            {tokensList}
          </div>
        </div>
      </div>
    </>
  );
}

export async function getStaticProps() {
  const fetcher = (path) =>
    fetch(
      `https://raw.githubusercontent.com/gsoft-inc/ov-igloo-tokens/main${path}`
    ).then((res) => res.json());

  const tokens = await fetcher('/docs/tokens.json');
  const tokensBase10 = await fetcher('/docs/base10/tokens.json');

  const categories = tokens.map((obj) => Object.keys(obj)).flat(1);

  if (!tokens || !tokensBase10) {
    return {
      notFound: true,
    };
  }

  return {
    props: { data: { default: tokens, base10: tokensBase10 }, categories }, // will be passed to the page component as props
  };
}
