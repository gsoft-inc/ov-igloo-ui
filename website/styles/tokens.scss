@use "base/colors" as color;
@use "abstract/function" as *;

// active color theme
$active-theme: color.$light-theme;

//space tokens
$space-0: var(--space-0);
$space-1: var(--space-1);
$space-2: var(--space-2);
$space-3: var(--space-3);
$space-4: var(--space-4);
$space-5: var(--space-5);
$space-6: var(--space-6);
$space-7: var(--space-7);
$space-8: var(--space-8);
$space-9: var(--space-9);
$space-10: var(--space-10);

//color tokens
$grey-100: var(--neutral-100);
$grey-200: var(--neutral-200);
$grey-300: var(--neutral-300);
$grey-400: var(--neutral-400);
$grey-500: var(--neutral-500);
$grey-600: var(--neutral-600);
$grey-700: var(--neutral-700);
$grey-800: var(--neutral-800);
$grey-900: var(--neutral-900);
$white: var(--neutral-0);
$black: var(--neutral-1000);
$green: #09a198;
$yellow: #ffc737;
$pink: #ff8ea9;

//radius tokens
$radius-0: var(--space-0);
$radius-1: rebase(0.5rem);
$radius-2: rebase(0.25rem);

// layout tokens
$layout-medium: rebase(80rem);
$layout-small: rebase(50rem);

// Semantic colors
$text-color: $black;
$bg-color: $white;
$body-bg-color: $grey-100;

$navbar-bg-color: $white;
$navbar-color: $black;
$navbar-min-height: var(--space-9);

$hero-space: $space-10 0 rebase(8rem) 0;

$menu-bg-color: $white;
$menu-color: $black;
$menu-border-color: $black;
$menu-border: 1px solid $menu-border-color;
$menu-text-size: 1.6rem;
$menu-radius: $radius-0;
$menu-shadow: 0.5px 1.6px 1.6px rgb(0 0 0 / 15%);
$code-bg-color: var(--code-300);
$code-color: var(--code-700);
$button-copy-border: rgba(205, 217, 229, 0.2);
$button-copy-color: #a7adb3;

// Semantic font sizes
$text-size: var(--fs-300);
$text-size-small: var(--fs-100);

$max-layout-width: $layout-medium;
