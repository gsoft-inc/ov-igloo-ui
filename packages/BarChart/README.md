# BarChart

Bar charts should be used when you are showing segments of information.

<Example is="custom" />

<ReferenceLinks is="custom" />

## Installation

To install `@igloo-ui/bar-chart` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/bar-chart
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/bar-chart
```

## Usage

Then to use the component in your code just import it!

```jsx
import BarChart from '@igloo-ui/bar-chart';

const data = [
    {
      label: 'Good vibes',
      value: 48,
      color: '#FBC73B',
    },
    {
      label: 'Excellence',
      value: 46,
      color: '#6893FF',
    },
]

<BarChart dataSet={data} />
```
