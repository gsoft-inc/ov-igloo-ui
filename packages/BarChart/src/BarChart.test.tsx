/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';

import BarChart from './BarChart';
import tokens from '@igloo-ui/tokens/dist/base10/tokens.json';

const setup = (props = {}) => {
  const mockData = [
    {
      label: 'Good vibes',
      value: 48,
      color: tokens.dandelion400,
    },
    {
      label: 'Excellence',
      value: 46,
      color: tokens.electricBlue500,
    },
    {
      label: 'Radical candor',
      value: 24,
      color: tokens.strawberryFields200,
    },
    {
      label: 'Custom cards by members',
      value: 0,
      color: tokens.sky100,
    },
  ];

  return render(
    <BarChart dataSet={mockData} dataTest="ids-bar-chart" {...props}>
      Hello world
    </BarChart>
  );
};

describe('BarChart', () => {
  test('It should render without errors', () => {
    setup();
    const wrapper = screen.getByTestId('ids-bar-chart');
    expect(wrapper).toBeInTheDocument();
  });

  test('It should render a snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });
});
