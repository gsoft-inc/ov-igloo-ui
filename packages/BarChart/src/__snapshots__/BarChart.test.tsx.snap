// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart It should render a snapshot 1`] = `
<DocumentFragment>
  <ul
    class="ids-bar-chart__container"
    data-test="ids-bar-chart"
  >
    <li
      class="ids-bar-chart"
    >
      <span
        class="ids-bar-chart__label"
      >
        Good vibes
      </span>
      <div
        class="ids-bar-chart__content"
      >
        <div
          class="ids-bar-chart__graph"
          data-value="48"
          style="width: 100%; background: rgb(251, 199, 59);"
        />
        <span
          class="ids-bar-chart__value"
        >
          48
        </span>
      </div>
    </li>
    <li
      class="ids-bar-chart"
    >
      <span
        class="ids-bar-chart__label"
      >
        Excellence
      </span>
      <div
        class="ids-bar-chart__content"
      >
        <div
          class="ids-bar-chart__graph"
          data-value="46"
          style="width: 95.83333333333333%; background: rgb(16, 83, 255);"
        />
        <span
          class="ids-bar-chart__value"
        >
          46
        </span>
      </div>
    </li>
    <li
      class="ids-bar-chart"
    >
      <span
        class="ids-bar-chart__label"
      >
        Radical candor
      </span>
      <div
        class="ids-bar-chart__content"
      >
        <div
          class="ids-bar-chart__graph"
          data-value="24"
          style="width: 50%; background: rgb(231, 145, 154);"
        />
        <span
          class="ids-bar-chart__value"
        >
          24
        </span>
      </div>
    </li>
    <li
      class="ids-bar-chart"
    >
      <span
        class="ids-bar-chart__label"
      >
        Custom cards by members
      </span>
      <div
        class="ids-bar-chart__content"
      >
        <div
          class="ids-bar-chart__graph"
          data-value="0"
          style="background: rgb(178, 221, 242);"
        />
        <span
          class="ids-bar-chart__value"
        >
          0
        </span>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
